src/compiler/emitter.ts

```diff
@@ -5650,16 +5702,20 @@ export function createPrinter(printerOptions: PrinterOptions = {}, handlers: Pri
      * Returns a value indicating whether a name is unique globally, within the current file,
      * or within the NameGenerator.
      */
-    function isUniqueName(name: string): boolean {
-        return isFileLevelUniqueName(name)
-            && !generatedNames.has(name)
-            && !(reservedNames && reservedNames.has(name));
+    function isUniqueName(name: string, privateName: boolean): boolean {
+        return isFileLevelUniqueName(name, privateName)
+            && !isReservedName(name, privateName)
+            && !generatedNames.has(name);
+    }
+
+    function isReservedName(name: string, privateName: boolean): boolean {
+        return privateName ? !!reservedPrivateNames?.has(name) : !!reservedNames?.has(name);
     }
 
     /**
      * Returns a value indicating whether a name is unique globally or within the current file.
      */
-    function isFileLevelUniqueName(name: string) {
+    function isFileLevelUniqueName(name: string, _isPrivate: boolean) {

```

Comment:
Is `_isPrivate` leftover from something? It seems like it could be removed if `isUniqueName` didn't pass it in.

Reply:
`isFileLevelUniqueName` is passed as a callback argument. The parameter isn't strictly necessary, but avoids v8 having to create an adaptor frame to handle the mismatched argument count. Adaptor frames [aren't nearly as expensive now](https://v8.dev/blog/adaptor-frame) as they were a few years back, but I'd still rather avoid them if possible, especially for a function that is potentially called frequently during downlevel emit.

Reply:
I'm confused; the only call to `isFileLevelUniqueName` is in `isUniqueName` and it's not passed as a callback argument anywhere, only called directly.

Is this just that `isUniqueName` has two parameters, so something special happens because it only calls other functions with two parameters?

Reply:
Oh, it's a call outside this diff, then?

Reply:
Yeah, `makeFileLevelOptimisticUniqueName`, gotcha.

(Might be worth a comment, but I see it now.)